'submodules' rule is not really useful

`make submodules` ([defined here](https://github.com/SpotlightKid/cookiecutter-dpf-effect/blob/6cafed22fe5055828c371e43630ea2cc884a8c1c/%7B%7B%20cookiecutter.repo_name%20%7D%7D/Makefile#L13-L14)) will fail if submodules have not yet been cloned, which kinda defeats the purpose of this rule...

```
$ make submodules
Makefile:7: dpf/Makefile.base.mk: No such file or directory
make: *** No rule to make target 'dpf/Makefile.base.mk'.  Stop.
```

Should we get rid of this rule? Otherwise, we could handle the "missing `dpf/Makefile.base.mk`" error more gracefully, and print an error message when submodules are missing:

```
DPF not found. Please run "make submodules" to clone the missing submodules, then retry building the plugin.
```

--- a/README.md
+++ b/README.md
@@ -136,7 +136,7 @@ variables differ depending on the target OS.*
 Requirements
 ------------
 
-* Basic development tools (C++ compiler, make, pkg-config, etc.)
+* Basic development tools (C++ compiler, GNU make, pkg-config, etc.)
 * Python
 * Git
 * [cookiecutter]

--- a/{{ cookiecutter.repo_name }}/Makefile
+++ b/{{ cookiecutter.repo_name }}/Makefile
@@ -4,7 +4,22 @@
 # Created by falkTX, Christopher Arndt, and Patrick Desaulniers
 #
 
-include dpf/Makefile.base.mk
+# error out if DPF is missing, unless the current rule is 'submodules'
+define MISSING_SUBMODULES_ERROR
+=============================================================================
+DPF library not found in directory 'dpf'.
+Please run "make submodules" to clone the missing Git submodules, then retry.
+=============================================================================
+endef
+
+ifneq ($(MAKECMDGOALS), submodules)
+ifeq (,$(wildcard dpf/Makefile.base.mk))
+    $(info $(MISSING_SUBMODULES_ERROR))
+    $(error Unable to continue)
+else
+    include dpf/Makefile.base.mk
+endif
+endif
 
 # --------------------------------------------------------------
 # Installation directories
@@ -78,16 +93,16 @@ plugins: libs
 
 ifneq ($(CROSS_COMPILING),true)
 gen: plugins dpf/utils/lv2_ttl_generator
-	@$(CURDIR)/dpf/utils/generate-ttl.sh
+	@dpf/utils/generate-ttl.sh
 ifeq ($(MACOS),true)
-	@$(CURDIR)/dpf/utils/generate-vst-bundles.sh
+	@dpf/utils/generate-vst-bundles.sh
 endif
 
 dpf/utils/lv2_ttl_generator:
 	$(MAKE) -C dpf/utils/lv2-ttl-generator
 else
 gen: plugins dpf/utils/lv2_ttl_generator.exe
-	@$(CURDIR)/dpf/utils/generate-ttl.sh
+	@dpf/utils/generate-ttl.sh
 
 dpf/utils/lv2_ttl_generator.exe:
 	$(MAKE) -C dpf/utils/lv2-ttl-generator WINDOWS=true

